### ISSUE TITLE
ImmutableRoaringBitmap.EMPTY ?

### ISSUE THREAD

**Issue by maciej** [RoaringBitmap/RoaringBitmap#65]
In some scenarios it is useful to have a _null object_ empty `ImmutableRoaringBitmap`. That would eliminate having to evaluate a null condition conditions on `ImmutableRoaringBitmap.or(ImmutableRoaringBitmap, ImmutableRoaringBitmap)`, `#and(ImmutableRoaringBitmap, ImmutableRoaringBitmap)` and other operations.
Do you think it makes sense?

**Comment by lemire** [comment]
Can you elaborate?

I think that what you are saying is that you would want "null" to be treated as if it were the empty set. Right?

**Comment by maciej** [comment]
In our use-case we have in the order of 10-100k bitmaps. Some of them are dense (and long), some sparse, but there are also a few that are empty. We'd like to avoid serialising the empty ones, only serialising those that have at least element.
Since `ImmutableRoaringBitmap` does not have a static `#bitmapOf(int...)` method as `RoaringBitmap` does, we'd have to maintain a `ByteBuffer` containing an empty `ImmutableRoaringBitmap` on the side.

```scala
val someBitmap = new ImmutableRoaringBitmap(bb)
// … unserialize a couple of more bitmaps

val anEmptyBitmap = ImmutableRoaringBitmap.EMPTY
// … fill in the blanks with EMTPY bitmaps
```
Is is more clear now?

Now when explaining this, I think that simply adding a `bitmapOf` method to ImmutableRoaringBitmap would be the optimal solution. What do you think?

**Comment by lemire** [comment]
> Since `ImmutableRoaringBitmap` does not have a static `#bitmapOf(int...)` method as `RoaringBitmap` does

MutableRoaringBitmap has this method:

https://github.com/RoaringBitmap/RoaringBitmap/blob/master/src/main/java/org/roaringbitmap/buffer/MutableRoaringBitmap.java#L155

So you can do ``(ImmutableRoaringBitmap) MutableRoaringBitmap.bitmapOf(...)`` does it help?

**Comment by maciej** [comment]
Thank you! I haven't realised that `ImmutableRoaringBitmap` is above the mutable one in the class hierarchy!

However, I'm still wondering if a convenience method doing exactly what you wrote:
```java
public static ImmutableRoaringBitmap bitmapOf(final int... dat) {
    return (ImmutableRoaringBitmap) MutableRoaringBitmap.bitmapOf(dat);
}
```

on `ImmutableRoaringBitmap` wouldn't make sense. I think it's not obvious that an `ImmutableRoaringBitmap` can be created with a `MutableRoaringBitmap.bitmapOf`. :-)

**Comment by lemire** [comment]
What you describe is possible and maybe a good idea, but I wonder: why is it important to you that the result be an ``ImmutableRoaringBitmap``? The largest design issue here is that ``MutableRoaringBitmap`` and ``ImmutableRoaringBitmap`` are interoperable. If this is not clear, then that is maybe the point we need to clarify in the API.

I thought this this was clear in the API doc...

http://static.javadoc.io/org.roaringbitmap/RoaringBitmap/0.5.19/org/roaringbitmap/buffer/ImmutableRoaringBitmap.html

No?

**Comment by maciej** [comment]
Let me try to explain this differently.
Both `RoaringBitmap` and `MutableRoaringBitmap` have a `bitmapOf` method. `ImmutableRoaringBitmap` does not.
For someone using the library for the first time (like me a couple of weeks ago) that's something surprising and confusing at first. Sure, looking at the documentation help, but I think the API would benefit from being a bit more consistent here. :-)

**Comment by lemire** [comment]
@maciej 

Ok.

**Comment by lemire** [comment]
Fixed in version 0.6.2.

Please upgrade. 

Reopen if it is not satisfying.
